### ISSUE TITLE
StructError failures() iterable only returns first error

### ISSUE THREAD

**Issue by rossholdway** [ianstormtaylor/superstruct#394]
When using `.failures()` to iterate over all errors (using the latest release, v0.10.11. Node v12.16.1) it only returns the first error.

I'd expect 2 errors to be logged below...

```javascript
const Article = object({
  id: number(),
  title: number()
})

const data = {
  id: 'a string',
  title: 'also a string'
}

const [error, article] = validate(data, Article);

for(const failure of error.failures()) {
  console.log(failure);
}

// Output
// {
//   value: 'a string',
//   type: 'number',
//   path: [ 'id' ],
//   branch: [ { id: 'a string', title: 'also a string' }, 'a string' ]
// }
```

**Comment by reppinR1** [comment]
It would be awesome if it got fixed.
Btw, loving your library!

**Comment by zenflow** [comment]
It would indeed be awesome if this limitation was fixed. 

Seems like this is the only feature that some of the alternatives have (e.g. yup, ajv) but superstruct does not.

**Comment by zenflow** [comment]
Submitted a PR to fix this: #440
